### ISSUE TITLE
Questions about SPDM message read/write between requester and responder

### ISSUE THREAD

**Issue by owen126315** [DMTF/spdm-emu#244]
I am trying to understand how requester send SPDM message over MCTP to responder through using wireshark to capture the tcp packets between requester and responder.

And I found that the captured packets seem mismatched from the SPDM message format and MCTP message format.

Is `SOCKET_SPDM_COMMAND_NORMAL` necessary for SPDM request/respond? 
Why the write function would split the request into four packets?
Why the mctp header information is not included in the packet payload?

https://github.com/DMTF/spdm-emu/blob/c8ac4ca68137127596ff59d065bfaa046718a56b/spdm_emu/spdm_emu_common/command.c#L258-L315

**Comment by jyao1** [comment]
yes, you are right. In current implementation, we just add MCTP message header. There is no MCTP transport header. The purpose of the SPDM-EMU is to test SPDM, not MCTP.

Feel free to proposal a patch to add MCTP transport header, if you see there is such need.

**Comment by FangYuFan** [comment]
> 

Hi Yao，
Do you know where the official MCTP protocol source code library is located? Because I couldn't find any relevant repository under the DMTF organization

**Comment by FangYuFan** [comment]
> yes, you are right. In current implementation, we just add MCTP message header. There is no MCTP transport header. The purpose of the SPDM-EMU is to test SPDM, not MCTP.
> 
> Feel free to proposal a patch to add MCTP transport header, if you see there is such need.

Hi Yao，
Do you know where the official MCTP protocol source code library is located? Because I couldn't find any relevant repository under the DMTF organization

**Comment by jyao1** [comment]
> Do you know where the official MCTP protocol source code library is located? Because I couldn't find any relevant repository under the DMTF organization

No. I don't know.
